src/facade/dragonfly_connection.cc

```diff
@@ -1530,7 +1532,7 @@ void Connection::SendInvalidationMessageAsync(InvalidationMessage msg) {
 }
 
 void Connection::LaunchDispatchFiberIfNeeded() {
-  if (!dispatch_fb_.IsJoinable()) {
+  if (!dispatch_fb_.IsJoinable() && !migration_in_process_) {
```

Comment:
This should break `SendAsync`

Reply:
in what sense?

Reply:
What I had in mind is that because we don't wake the dispatch fiber it would miss the `evc.notify` and when the fiber was launched again it would not consume the items that were pushed back on the dispatch queue. I am wrong, I double checked the flow and we guard against this case in the dispatch fiber's main loop:

```
  1340     evc_.await(
  1341         [this] { return cc_->conn_closing || (!dispatch_q_.empty() && !cc_->sync_dispatch); });
  1342     if (cc_->conn_closing)                                                                        
  1343       break;

```

Also a corner case that came into my mind, if after a migration the connection for some reason is `closing` are there any cases where we need to `drain` the `dispatch queue` ?

Reply:
ClearPipelinedMessages is called at the end and it drains the queue